ecl/agentexec/agentexec.cpp

```diff
@@ -342,13 +342,8 @@ class WaitThread : public CInterfaceOf<IPooledThread>
             Owned<IConstWorkUnit> cw = factory->openWorkUnit(wuid);
             if (cw)
             {
-                // if either a) NOT a thoragent with useChildProcesses=false (default in k8s config) or b) is still in an executing state
-                if (!sharedK8sJob || (cw->getState() == WUStateRunning) || (cw->getState() == WUStateBlocked) || (cw->getState() == WUStateWait))
+                if (!sharedK8sJob && (cw->getState() == WUStateRunning) || (cw->getState() == WUStateBlocked) || (cw->getState() == WUStateWait))
```

Comment:
I think you have the wrong precedence.  You need to have brackets around all the || conditions
a && (b || c || d)
otherwise it will be evaluated as
(a && b) || c || d

Also best for clarity...

Reply:
whoops, yes, missing brackets